core/dbt/parser/manifest.py

```diff
@@ -553,6 +555,18 @@ def is_partial_parsable(self, manifest: Manifest) -> Tuple[bool, Optional[str]]:
             logger.info("Unable to do partial parsing because profile has changed")
             valid = False
             reparse_reason = ReparseReason.profile_changed
+        if self.manifest.state_check.project_env_vars_hash != \
+                manifest.state_check.project_env_vars_hash:
+            logger.info("Unable to do partial parsing because env vars "
+                        "used in dbt_project.yml have changed")
+            valid = False
+            reparse_reason = ReparseReason.proj_env_vars_changed
+        if self.manifest.state_check.profile_env_vars_hash != \
+                manifest.state_check.profile_env_vars_hash:
+            logger.info("Unable to do partial parsing because env vars "
+                        "used in profiles.yml have changed")
```

Comment:
Depending on what goes in first (this or #4055) , this will need to be updated to the new struct logging as well as anywhere else logging is happening.  Nothing to do now - more of a reminder for later.

Reply:
I think we should have a walkthrough next week since everything is all crunchy right now.